src/librbd/TaskFinisher.h

```diff
@@ -45,6 +45,7 @@ class TaskFinisher {
     typename TaskContexts::iterator it = m_task_contexts.find(task);
     if (it != m_task_contexts.end()) {
       delete it->second.first;
+      m_safe_timer->cancel_event(it->second.second);
```

Comment:
Hmm -- so by erasing the task from ```m_task_contexts``` below, when the timer fires and invokes ```complete(const Task&)```, that method shouldn't find the associated task and thus it turns into a no-op.